### PR TITLE
feat: update devnet script for Ecotone upgrade compatibility

### PR DESCRIPTION
**Pull request by 0x6e616d**
Update `devnet` script to for Ecotone upgrade compatibility

### CODE REVIEW

File: bedrock-devnet/devnet/__init__.py [hunk 1]
```diff
@@ -401,18 +365,6 @@ def eth_new_head(url):
...
-    headers = {'Content-type': 'application/json'}
-    body = '{"id":3, "jsonrpc":"2.0", "method": "debug_dumpBlock", "params":["latest"]}'
-    conn.request('POST', '/', body, headers)
-    response = conn.getresponse()
-    data = response.read().decode()
-    conn.close()
-    return data
-
-
 def wait_for_rpc_server(url):
```
Comment (0x6e616d) [inline bedrock-devnet/devnet/__init__.py:368]: unused function

File: bedrock-devnet/devnet/__init__.py [hunk 2]
```diff
@@ -548,9 +500,6 @@ def validate_fork_public_network(args):
       log.info(f'Fork from RPC URL: {l1_rpc_url}, block number: {block_number}, l2 native token: {l2_native_token}')
 
 
-def pad_hex(input):
-    return '0x' + input.replace('0x', '').zfill(64)
-
```
Comment (0x6e616d) [inline bedrock-devnet/devnet/__init__.py:553]: unused function

File: bedrock-devnet/devnet/__init__.py [hunk 3]
```diff
@@ -157,46 +154,6 @@ def main():
...
-    run_command(cmd, env={}, cwd=paths.contracts_bedrock_dir)
-
-    shutil.copy(paths.l1_deployments_path, paths.addresses_json_path)
-
-    log.info('Syncing contracts.')
-    run_command([
-        'forge', 'script', fqn, '--sig', 'sync()',
-        '--memory-limit', '268435456',
-        '--rpc-url', 'http://127.0.0.1:8545'
-    ], env={}, cwd=paths.contracts_bedrock_dir)
```
Comment (0x6e616d) [inline bedrock-devnet/devnet/__init__.py:199]: unused function

File: bedrock-devnet/devnet/__init__.py [hunk 4]
```diff
@@ -9,9 +9,6 @@
 import shutil
 import http.client
 import gzip
-import glob
-import ssl
-import urllib
```
Comment (0x6e616d) [inline bedrock-devnet/devnet/__init__.py:14]: Unused imported libs

File: bedrock-devnet/devnet/__init__.py [hunk 5]
```diff
@@ -342,27 +299,34 @@ def devnet_deploy(paths, args):
...
     log.info(f'Using batch inbox {batch_inbox_address}')
-
-    log.info('Bringing up `op-node`, `op-proposer` and `op-batcher`.')
-    run_command(['docker', 'compose', 'up', '-d', 'op-node', 'op-proposer', 'op-batcher'], cwd=paths.ops_bedrock_dir, env={
+    # Set up the base docker environment.
+    docker_env={
         'PWD': paths.ops_bedrock_dir,
-        'L2OO_ADDRESS': l2_output_oracle,
         'SEQUENCER_BATCH_INBOX_ADDRESS': batch_inbox_address,
+        'L2OO_ADDRESS': l2_output_oracle,
```
Comment (nguyenzung) [inline bedrock-devnet/devnet/__init__.py:306]: Maybe we don't need 'L2OO_ADDRESS' here. We have another check at: https://github.com/tokamak-network/tokamak-thanos/blob/feat/OR-1619-devnet-ecotone/bedrock-devnet/devnet/__init__.py#L321

Comment (0x6e616d) [reply bedrock-devnet/devnet/__init__.py]: That's true, updated

**Review by rlgns98kr — APPROVED** [review]
Thank you very much! 🙇🏻‍♂️

**Pull request merged**